src/config/settings.ts

```diff
@@ -0,0 +1,53 @@
+import path from "node:path";
+import os from "node:os";
+import * as vscode from "vscode";
+
+export type VSCodeSettings = {
+    configPath: string | undefined;
+};
+
+export const getVSCodeSettings = (wf?: vscode.WorkspaceFolder): VSCodeSettings => {
+    const rootConfig = vscode.workspace.getConfiguration("testplane");
+    const folderConfig = vscode.workspace.getConfiguration("testplane", wf);
+
+    const get = <T>(key: string, defaultValue?: T): T | undefined =>
+        getSettingValue<T>({
+            rootConfig,
+            folderConfig,
+            key,
+            defaultValue,
+        });
+
+    const configPath = get<string | undefined>("configPath");
```

Comment:
Do you need to specify `| undefined` here? `get` already returns `T | undefined`, so it should be `string | undefined` if you only specify `<string>`